### PR TITLE
Docs - 4414 - Add API Managed Table Story

### LINKED ISSUE #4414: Add a story file to /apiManagedTable demonstrating how to put pieces together
apiManagedTable holds a set of components that can be used together to produce a paginated table with features like searching, filtering, and exporting data. Add a story file which demonstrates with basic examples how these components can be assembled in different ways. This may make it easier for other devs to use these components to build new tables.

### PR DESCRIPTION
**Pull request by esizer**
## 👋 Introduction

This adds a very basic story for the `apiManagedTable` in admin.

## 🧪 Testing

> **Warning**
> Search is not functional, do we need it be? I tried to get it working and it just seems like a a lot of effort for a story. However, if we believe this is critical we can bump up the estimate and I'll continue working on getting this in.

1. Review chromatic
2. Confirm story "Api Managed Table" exists and displays as expected

## 🤖 Robot Stuff

Resolves #4414

### CODE REVIEW

File: frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+};
+
+type Data = NonNullable<FromArray<Users>>;
+
+const columns: ColumnsOf<Data> = [
+  {
+    label: "First Name",
+    id: "name",
+    accessor: (row: User) => row.firstName,
+    sortColumnName: "title",
```
Comment (petertgiles) [inline frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx:46]: Should be firstName? (Also with lastName?)  It makes both columns show the sort icon together.

**Review by yonikid15 — APPROVED** [review]
LGTM 👍 
I found some bugs that are out of scope for this:
- Clicking columns button breaks the story. However it seems to happening across all tables in storybook. Error: `TypeError: Cannot read properties of null (reading 'control')`
- Small bug when changing the number of items in the pagination. To recreate bug: go to page 10, click "Show 50" in select field option. You'll see no users show up since the page selector (number input) is 'stuck' on page 10.

I'll make new issues for these.

**Comment by esizer** [comment]
> * Clicking columns button breaks the story. However it seems to happening across all tables in storybook. Error: `TypeError: Cannot read properties of null (reading 'control')`

Yup, this was fixed with a recent merge. 👍 

> * Small bug when changing the number of items in the pagination. To recreate bug: go to page 10, click "Show 50" in select field option. You'll see no users show up since the page selector (number input) is 'stuck' on page 10.

I know exactly what is going on here and will fix it before merging. Thanks!

File: frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  return (
+    <>
+      <h2 id="heading">API Managed Table</h2>
+      <p>Note: Search not functional</p>
+      <TableHeader
+        columns={columns}
+        hiddenColumnIds={hiddenColumnIds}
+        filterButtonComponent={null}
+        onSearchChange={(term, type) => {
+          action(JSON.stringify({ term, type }));
```
Comment (petertgiles) [inline frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx:115]: I don't think I'm getting anything out of this.  Are you?

**Review by petertgiles — CHANGES_REQUESTED** [review]
This is cool!  A few things to check on but I don't think it needs to actually work.  Showing activity in the actions panel is sufficient, I think.

File: frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx (hunk 2 above)
Comment (esizer) [reply frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx]: Did not realize I had to wrong syntax. I think we may need to review our story actions cause I just copied an existing one.

Fixed in [fix search action logging](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4542/commits/b50f888452f8ed6b8e7ee73da94f5fbf91fd4d8c)

File: frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  }, [pageSize, currentPage]);
+
+  return (
+    <>
+      <h2 id="heading">API Managed Table</h2>
+      <p>Note: Search not functional</p>
+      <TableHeader
+        columns={columns}
+        hiddenColumnIds={hiddenColumnIds}
+        filterButtonComponent={null}
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx]: Excellent idea, added in [add placeholder filter button](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4542/commits/55dc9d5bb0c10201dbf66cdf4a6066160f099d63)

File: frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx (hunk 1 above)
Comment (esizer) [reply frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx]: Oops, copy and paste error :P

Fixed with [fix column id and sort name](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4542/commits/1268cc881f1e38842eb42b6b3ee7e689047b5c2d)

File: frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    return mockUsers.slice(start, end);
+  }, [pageSize, currentPage]);
+
+  return (
+    <>
+      <h2 id="heading">API Managed Table</h2>
+      <p>Note: Search not functional</p>
+      <TableHeader
+        columns={columns}
+        hiddenColumnIds={hiddenColumnIds}
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx:113]: Yeah, sorry. This was a bug that was fixed in another PR. I merged main and it is now working again 👍

File: frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+            event,
+          );
+        }}
+        {...header}
+      />
+      <BasicTable
+        labelledBy="heading"
+        columns={columns}
+        data={paginatedUsers}
+        onSortingRuleChange={setSortingRule}
```
Comment (petertgiles) [inline frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx:142]: Could we get an action logged for sort changes, too?

**Review by petertgiles — CHANGES_REQUESTED** [review]
Looks lovely!  Could I trouble you for one more thing?

File: frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx (hunk 5 above)
Comment (esizer) [reply frontend/admin/src/js/components/apiManagedTable/ApiManagedTable.stories.tsx]: Add actions to all events in [add more actions for events](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4542/commits/d45e569b860e36bb1fc44bd1928db0f9bc249fce).

**Pull request merged**